packages/snaps-controllers/src/cronjob/CronjobController.ts

```diff
@@ -267,24 +335,131 @@ export class CronjobController extends BaseController<
   }
 
   /**
-   * Unregister all jobs related to the given snapId.
+   * Schedule a background event.
+   *
+   * @param backgroundEventWithoutId - Background event.
+   * @returns An id representing the background event.
+   */
+  scheduleBackgroundEvent(
+    backgroundEventWithoutId: Omit<BackgroundEvent, 'id' | 'scheduledAt'>,
+  ) {
+    const event = {
+      ...backgroundEventWithoutId,
+      id: nanoid(),
+      scheduledAt: new Date().toISOString(),
+    };
+    this.#setUpBackgroundEvent(event);
+    this.update((state) => {
+      state.events[event.id] = castDraft(event);
+    });
+
+    return event.id;
+  }
+
+  /**
+   * Cancel a background event.
+   *
+   * @param id - The id of the background event to cancel.
+   * @param origin - The origin making the cancel call.
+   * @throws If the event does not exist.
+   */
+  cancelBackgroundEvent(id: string, origin: string) {
```

Comment:
It'll likely be easier to bind this if we move the `origin` param to be the first argument